File: UserManagement.Web/Models/BillDetailModel.cs
using UserManagement.Web.Helpers;
using System;
using System.Collections.Generic;

namespace UserManagement.Web.Models
{
    public class BillDetailModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Phone { get; set; }
        public string Email { get; set; }
        public decimal? Tax { get; set; }
        public decimal TotalAmount { get; set; }
        public decimal? Discount { get; set; }
        public string Remark { get; set; }
        public Constants.BillStatus Status { get; set; }
        public DateTime CreatedOn { get; set; }

        public VendorPersonalInfoModel Vendor { get; set; }

        public IEnumerable<ServiceModel> Items { get; set; }
        public IEnumerable<AttachmentModel> Attachments { get; set; }
    }
}

File: UserManagement.Web/ClientApp/src/models/bill/bill.detail.model.ts
import { AttachmentDetailModel } from '../attachment/attachment.detail.model';
import { VendorPersonalInfoModel } from '../vendor/vendor.personal.info.model';
import { SalesTaxDetailModel } from '../sales-tax/sales.tax.detail.model';

export class BillDetailModel {
    public id: number;
    public vendorId: string;
    public name:string;
    public phone: string;
    public email: string;
    public referenceNumber: string;
    public dueDate: string;
    public tax: number;
    public discount: number;
    public totalAmount: number;
    public remark: string;
    public status: number;
    public lineAmountSubTotal: number;

    public vendor: VendorPersonalInfoModel;

    public items: any;
    public attachments: Array<AttachmentDetailModel>;

    constructor() {
        this.vendor = new VendorPersonalInfoModel();

        this.items = new Array<number>();
        this.attachments = new Array<AttachmentDetailModel>();
    }
}
